spec/controllers/users/registrations_controller_spec.rb

```diff
@@ -0,0 +1,50 @@
+require 'rails_helper'
+
+describe Users::RegistrationsController, type: :request do
+
+  let (:user) { build_user }
+  let (:existing_user) { create_user }
+  let (:signup_url) { '/signup' }
+  let(:params) do
+    {
+      user: {
+        email: user.email,
+        password: user.password
+      }
+    }
+  end
+
+  context 'When creating a new user' do
+    before do
+      post signup_url, params: params
+    end
+
+    it 'returns 200' do
+      expect(response.status).to eq(200)
+    end
+
+    it 'returns a token' do
+      expect(response.headers['Authorization']).to be_present
+    end
+
+    it 'returns the user email' do
+      expect(json["email"]).to eq(user.email)
+    end
+  end
+
+  context 'When an email already exists' do
+    before do
+      post signup_url, params: {
+        user: {
+          email: existing_user.email,
+          password: existing_user.password
+        }
+      }
+    end
+
+    it 'returns 422' do
+      expect(response.status).to eq(422)
```

Comment:
@josephelias-rubyians  Lets also return a message in json response something like user already exists.

Reply:
yes devise is returning message like "Email already taken"

Reply:
okay. Lets write a spec for the error response messages.